src/features/lightspeed/utils/updateRolesContext.ts

```diff
@@ -0,0 +1,97 @@
+import * as path from "path";
+import * as fs from "fs";
+
+import {
+  IVarsContext,
+  IWorkSpaceRolesContext,
+} from "../../../interfaces/lightspeed";
+import { readVarFiles } from "./readVarFiles";
+import { VarType } from "../../../interfaces/lightspeed";
+
+function getVarsFromRoles(
+  rolePath: string,
+  varType: VarType
+): IVarsContext | undefined {
+  const varsRootPath = path.join(rolePath, varType);
+  if (!fs.existsSync(varsRootPath)) {
+    return;
+  }
+  const varsFiles =
+    fs
+      .readdirSync(varsRootPath)
```

Comment:
The bug found was caused by a regular file named `roles`.  Can't a regular file `vars` cause a similar problem here? I think we'd better add try/catch block here as well.

Reply:
Yes! You're right. Let's me cover this case.